src/containers/course-section/resources-list/ResourcesList.tsx

```diff
@@ -56,6 +56,21 @@ const ResourcesList: FC<ResourcesListProps> = ({
     setResources(reorderedItems)
   }
 
+  const setResourceAvailability = (
+    id: string,
+    availability: ResourceAvailabilityStatus,
+    openFromDate?: string | null
+  ) => {
+    const resources = [...items]
+    const resource = resources.find((item) => item._id === id)
```

Comment:
you can use items.find and below use items in if statement, so resources variable is redundant, also, you can rename items prop to resources

Reply:
I can't directly change element of array in useState

Reply:
you can use:
setResources([...resources])

and in other places where you are using resources you can just use items, so, you will avoid to create the resources variable

Reply:
ok, but I don`t see the place where it can set undefined to openFromDate, as I see according to code and interface in ResourceItem where you passed this function, openFromDate can be null or string but not undefined, or I am wrong?

Reply:
I cannot get the required item using items.find because items is a prop and I cannot directly make changes to the found object. It will not update at all.